SafetyFirst/ResultModule.fs

```diff
@@ -23,6 +33,20 @@ module ResultExpression =
   /// </summary>
   let result = new ResultExpression ()
 
+  /// <summary>
+  /// The computation expression to manage Result as an Applicative Functor.
+  ///
+  /// Code wrapped in a Result is known to potentially fail with specific errors.
+  /// 
+  /// Use this computation expression in place of <c>result</c> when:
+  /// 1. None of the results depend on each other, and all are independent.
```

Comment:
what happens if this isn't the case? Do you just get extra failures?

Reply:
It's builtin to the syntax.  If I say
```f# 
result { 
  let! a = f x
  let! b = g a
  return b + 2
}
```
that's fine, but if you say
```f# 
validate (+) { 
  let! a = f x
  and! b = g a
  return b + 2
}
```
it'll fail to compile because `a` isn't in scope on your `and!` line.

Reply:
would you get a compile error for
```
validate (+) { 
  let! a = f x
  let! b = g a
  return b + 2
}
```
?

Reply:
![image](https://user-images.githubusercontent.com/15835006/132058888-ddda15be-786b-4d45-84fd-ba53d1603935.png)

Reply:
> would you get a compile error for
> 
> ```
> validate (+) { 
>   let! a = f x
>   let! b = g a
>   return b + 2
> }
> ```
> 
> ?

yup
![image](https://user-images.githubusercontent.com/15835006/132059365-60b8e916-139b-426b-a3c7-9c44e70b8107.png)

Reply:
That's potentially an argument for waiting until we put FSharpPlus as a dependency in this library, and use regular monoid/semigroup append instead of supplying our own.  I'm pretty sure we could then merge this CE with the normal `result` CE.  So if you said
```F#
result { 
  let! a = Error "1"

  let! b = Error "2"
  and! c = Error "3"

  return a + b + c
}
```
it would compile and give you `Error "1"`.  But if you had
```F#
result { 
  let! a = Ok 1

  let! b = Error "2"
  and! c = Error "3"

  return a + b + c
}
```
you'd get `Error "23"`.  Plus the naming would be cleaner.  But you get less flexibility, like if you want to have strings that get appended together using a custom function (like I have where it puts a new line between each one), you could only do that by defining an SCU with it's own semigroup instance. (At that point, you'd probably just make it a list of strings and fold the list together instead)

Reply:
Indeed.  See #84.  Any preference?